Xamarin.Forms.Platform.UAP/FontImageSourceHandler.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Graphics.Canvas;
+using Microsoft.Graphics.Canvas.Text;
+using Windows.Storage.Streams;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Xamarin.Forms.Platform.UWP
+{
+	public sealed class FontImageSourceHandler : IImageSourceHandler
+	{
+		public async Task<Windows.UI.Xaml.Media.ImageSource> LoadImageAsync(ImageSource imagesource,
+			CancellationToken cancelationToken = default(CancellationToken))
+		{
+			BitmapSource image = null;
+			if (imagesource is FontImageSource fontsource)
+			{
+				var iconcolor = fontsource.Color != Color.Default ? fontsource.Color : Color.White;
+				var device = CanvasDevice.GetSharedDevice();
+				var localDpi = Windows.Graphics.Display.DisplayInformation.GetForCurrentView().LogicalDpi;
```

Comment:
Android and iOS are not so much, but UWP supports multi-scale and multi-monitor. The common case would be to move the app from one screen to another, potentially changing the resolution. We might want to handle this case to avoid distorted images.

Reply:
Ideally, yes. But here we return the bitmap. And if it is redrawn at each DPI measurement, then this greatly affects the performance. With a large number of icons, the application freezes for a couple of seconds. Therefore, I added a minimum dpi equal to 300. This should be enough so that the icon does not look soapy in most cases.